source/checkpoint_restart.f90

```diff
@@ -109,6 +110,75 @@ subroutine cr_expand_arrays(npart_new)
 
 end subroutine cr_expand_arrays
 
+! ================================================================================================ !
+!  CR KILL SWITCH
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Last modified: 2018-11-16
+!  This routine will kill SixTrack if the current turn number matches a number in crkillturns
+! ================================================================================================ !
+subroutine cr_killSwitch(iTurn)
+
+  use crcoall
+  use file_units
+  use mod_settings
+  use mod_meta
+
+  integer, intent(in) :: iTurn
+
+  logical killIt, fExist
+  integer pTurn, nKills, i
+
+  killIt = .false.
+
+  if(crksunit == -1) then
+    call funit_requestUnit("crkillswitch.tmp",crksunit)
+  end if
+
+  inquire(file="crkillswitch.tmp",exist=fExist)
+  if(fExist .eqv. .false.) then
+    open(crksunit,file="crkillswitch.tmp",form="unformatted",access="stream",status="replace")
```

Comment:
why don't you use `units_openUnit` directly?

Reply:
Because it's supposed to be used for global files for flushing and closing at end of simulation, and to properly interface with BOINC. Here the tmp file is opened and closed internally in the routine every time, and the files are not needed for anything other than temporary storage.